cmd/proxy/manager.go

```diff
@@ -346,6 +355,73 @@ func (t *Manager) ConfigureRouter(r *gin.Engine) {
 		}
 	})
 
+	r.GET("/api/v1/setup-status", func(c *gin.Context) {
+		c.Stream(func(w io.Writer) bool {
+			logfile := fmt.Sprintf("/root/network/logs/%s.log", t.network)
+			t, err := tail.TailFile(logfile, tail.Config{
+				Follow: true,
+				ReOpen: true})
+			if err != nil {
+				return false
+			}
+			for line := range t.Lines {
+				if strings.Contains(line.Text, "Waiting for XUD dependencies to be ready") {
+					status := SetupStatus{Status: "Waiting for XUD dependencies to be ready", Details: nil}
+					j, _ := json.Marshal(status)
+					c.Writer.Write(j)
+					c.Writer.Write([]byte("\n"))
+					c.Writer.Flush()
+				} else if strings.Contains(line.Text, "LightSync") {
+					parts := strings.Split(line.Text, " [LightSync] ")
+					parts = strings.Split(parts[1], " | ")
```

Comment:
This approach is OK for the initial release/MVP, but seems quite fragile. Do we have a way to make this approach more reliable in the future?

Reply:
@erkarl Yes. As you saw current implementation is based on parsing the log file. I think it's better to let utils to provide this API because the utils does such setup task. Currently the proxy(api) container is started with other containers at the same time. So this makes it less reasonable to present such setup-status API on proxy(api) container. In order to optimize it I came up to two solutions:

Solution 1. Start proxy(api) container first, then the others. And move setup logic into proxy(api) so that the proxy can provide setup-status without any log parsing.

Solution 2. Provide APIs in utils? And proxy this utils API on proxy(api) container. But this is the idea we tried to avoid at first.

BTW, solution 1 fits the future of xud-docker. We provided a small image (proxy) which could be run on most platforms. And it has APIs by default. So everything we did in utils before has its own API on proxy. Then utils can be retired.

Reply:
Solution 1 seems like the right direction to go in the long run (outside the scope of this pull request though).